src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeSession.cs

```diff
@@ -84,10 +95,26 @@ public async Task StopAsync(CancellationToken cancellationToken)
         private static IpcMessage CreateStartMessage(EventPipeSessionConfiguration config)
         {
             // To keep backward compatibility with older runtimes we only use newer serialization format when needed
-            // V3 has added support to disable the stacktraces
-            bool shouldUseV3 = !config.RequestStackwalk;
-            EventPipeCommandId command = shouldUseV3 ? EventPipeCommandId.CollectTracing3 : EventPipeCommandId.CollectTracing2;
-            byte[] payload = shouldUseV3 ? config.SerializeV3() : config.SerializeV2();
+            EventPipeCommandId command;
+            byte[] payload;
+            if (config.RundownKeyword.HasValue && config.RundownKeyword.Value != DefaultRundownKeyword && config.RundownKeyword.Value != 0)
```

Comment:
Is there actual value in checking that the rundown keyword is not the default rundown value? Even if it does have the default rundown value, what harm is there in using CollectTracing4? I think that this should only check if  `RundownKeyword` is not null and drop the rest of the conditions.

Reply:
The comment above explained why - this is done to maximize the probability of success. For some time to come people will get stuck on older runtimes (e.g. there stay on 8 until the next LTS or until 8 ran out of support).

There is no point in taking an unnecessary risk of failing (although failing doesn't *really* matter)

With my new iteration, `config.RundownKeyword` is no longer nullable, it should make things simpler.